islands/InfoModal.tsx

```diff
@@ -0,0 +1,303 @@
+                    data-tooltip-placement="top"
+                    className="mt-1 mr-1 text-white bg-transparent hover:bg-gray-500 hover:text-white-900 rounded-lg text-sm w-8 h-8 ml-auto inline-flex justify-center items-center dark:hover:bg-gray-600 dark:hover:text-white"
+                    data-modal-hide="accordion-collapse"
+                  >
+                    <IconPlus class="w-5 h-5 text-white" />
+                  </button>
+                  <div
+                    id="add-pipeline-tooltip"
+                    role="tooltip"
+                    class="absolute z-10 invisible inline-block px-2 py-1 text-xs font-medium text-white transition-opacity duration-300 bg-gray-500 rounded-xs shadow-sm opacity-0 tooltip dark:bg-gray-700"
+                  >
+                    Add new pipeline
+                    <div class="tooltip-arrow" data-popper-arrow></div>
+                  </div>
+                </a>
+              </div>
+              <button
+                id="attached-pipeline"
+                data-dropdown-toggle="attached-pipeline-dropdown"
+                class="text-white border border-gray-600 font-medium rounded-sm w-full flex justify-between text-sm px-2 text-xs py-1 text-center inline-flex items-center"
+                type="button"
+              >
+                {attachedPipeline?.name}
+                <svg
+                  class="w-2.5 h-2.5 ml-2.5"
+                  aria-hidden="true"
+                  xmlns="http://www.w3.org/2000/svg"
+                  fill="none"
+                  viewBox="0 0 10 6"
+                >
+                  <path
+                    stroke="currentColor"
+                    stroke-linecap="round"
+                    stroke-linejoin="round"
+                    stroke-width="2"
+                    d="m1 1 4 4 4-4"
+                  />
+                </svg>
+              </button>
+              <div
+                id="attached-pipeline-dropdown"
+                class="z-10 hidden bg-white divide-y divide-gray-100 rounded-sm shadow w-[250px] w-full dark:bg-gray-700"
+              >
+                <ul
+                  class="py-2 text-sm text-gray-700 dark:text-gray-200"
+                  aria-labelledby="dropdownDefaultButton"
+                >
+                  {pipelines.map((pipeline: any) => (
+                    <li>
+                      <button
+                        data-tooltip-target={pipeline.id !==
+                            attachedPipeline?.id
+                          ? "tooltip-default"
+                          : null}
+                        data-tooltip-placement="top"
+                        type="button"
+                        class={`px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white flex justify-start text-left w-full ${
+                          pipeline.id === attachedPipeline?.id && "font-bold"
+                        }`}
+                        //todo pass down envars to browser
```

Comment:
You could create a route to confirm the attach...like `/service/[service]/component[component]/op/[op]/pipeline/[id]/attach` (I know it's a bit verbose). Then in that route you can render a confirm modal and then the modal can submit a form POST to the same route where you can have a POST handler do the attach. I will set up pipeline delete just like that you can use as an example.